OutboxSample/Application/IUserRepository.cs

```diff
@@ -4,13 +4,13 @@ namespace OutboxSample.Application;
 
 public interface IUserRepository : IRepository, ISupportUnitOfWork
 {
-    IEnumerable<User> GetAll();
+    QueryResult<IEnumerable<User>> GetAll();
 
-    User? Get(Guid id);
+    QueryResult<User> Get(Guid id);
 
-    bool Add(User user);
+    QueryResult<int> Add(User user);
 
-    bool AddMany(IEnumerable<User> users);
+    QueryResult<int> AddMany(IEnumerable<User> users);
```

Comment:
Is returning an `int` on `Add` and `AddMany` just to take advantage of the generic implementation of `QueryResult`?

Reply:
You could just as well return plain `QueryResult`. The in is to return the number of affected records. Not sure it'll have much use in this specific case, so this is just to show-case the opportunity,